blocks/header/headerutil.js

```diff
@@ -298,6 +298,108 @@ const processBondsType = (items) => {
   return processedData;
 };
 
+function loadStockFeed(gaTokenId) {
+  let socket = null;
+  const nifty50 = '4.1!NIFTY 50';
+  const sensex = '1.1!SENSEX';
+
+  // Get current date in IST timezone
+  const now = new Date();
+  const options = {
+    timeZone: 'Asia/Kolkata', hour: '2-digit', minute: '2-digit', weekday: 'long', hour12: false,
+  };
+  const formatter = new Intl.DateTimeFormat([], options);
+  const ISTTime = formatter.format(now);
+
+  // Extract day, time from ISTTime
+  const [day, ...timeParts] = ISTTime.split(' ');
+  const time = timeParts.join(' ');
+
+  // Extract hour, minute, and period from time
+  const [hourMinute] = time.split(' ');
+  const [hour, minute] = hourMinute.split(':').map(Number);
+
+  // Set feedFlag based on the time and day
+  const feedFlag = day !== 'Saturday' && day !== 'Sunday' && ((hour > 9 || (hour === 9 && minute >= 15)) && (hour < 15 || (hour === 15 && minute <= 30)));
+
+  const connect = (hostname, token) => {
+    // eslint-disable-next-line no-undef
+    socket = io.connect(hostname, {
+      auth: { token },
+      transports: ['websocket'],
+    });
+  };
+
+  const watch = (symbols) => {
+    socket.emit('join', symbols);
+  };
+
+  const onChange = (callback) => {
+    socket.on('stock', callback);
+  };
+
+  const getScript = (source, callback) => {
+    const script = document.createElement('script');
+    const prior = document.getElementsByTagName('script')[0];
+    script.async = 1;
+    script.src = source;
+    script.onload = callback;
+    prior.parentNode.insertBefore(script, prior);
+  };
+
+  function updateStockInfo(stockName, stockData) {
```

Comment:
The logic written here should be extracted at one place, for both initial API call and then subsiquent web socket updates. This common methos would take the stock values and update the elements data and classnames for both API calls

Reply:
the structure of API response and streaming message is completely different so accommodating both in one method will unnecessary complexity.

Reply:
I meant the following Structure:
StockObject {
Nifty: {...},
Sensex: {...}
} - Object with processes nifty and Sensex price, change percentage 

StockObject: getStockValuesFromAPI(); - Processes the stock API response and returns the StockObject

StockObject: getStockValuesFromStreamingAPI(); - Processes the socket API messages and returns the StockObject

decorateStockValues(StockObject); - Decorates/readers the dynamic stock section when the StockObject is passed in this function. This takes care of adding classes like positive, stock up, down etc on elements

Reply:
It is not that straight forward as at the time of streaming data the last stock value from the last stream need to be compared with the current one every time and decide the color of value while it is not needed at the time of API data, so just trying to have common method will not server any fruitful purpose IMO.

Reply:
okay got it, thanks